apps/issuance/src/issuance.service.ts

```diff
@@ -133,28 +144,37 @@ export class IssuanceService {
     }
   }
 
-
-  async sendCredentialOutOfBand(payload: OOBIssueCredentialDto): Promise<{ response: object; }> {
+  async sendCredentialOutOfBand(payload: OOBIssueCredentialDto): Promise<{ response: object }> {
     try {
       const { orgId, credentialDefinitionId, comment, attributes, protocolVersion } = payload;
 
-      const attrError = [];
-      if (0 < attributes?.length) {
-        attributes?.forEach((attribute, i) => {
-      
-            if (attribute.isRequired && !attribute.value) {
-              attrError.push(`attributes.${i}.Value of "${attribute.name}" is required`);
-              return true;
-            }
+      const schemadetailsResponse: SchemaDetails = await this.issuanceRepository.getCredentialDefinitionDetails(
+        credentialDefinitionId
+      );
+
+      if (schemadetailsResponse?.attributes) {
+        const schemadetailsResponseError = [];
+        const attributesArray: IAttributes[] = JSON.parse(schemadetailsResponse.attributes);
+
+        attributesArray.forEach((attribute) => {
+          if (attribute.attributeName && attribute.isRequired) {
             
-            return attribute.isRequired && !attribute.value;
-          });
-      
-        if (0 < attrError.length) {
-          throw new BadRequestException(attrError);
-        }
+            payload.attributes.map((attr) => { 
```

Comment:
@pranalidhanavade It would be better to create a common function for this logic and call from wherever required